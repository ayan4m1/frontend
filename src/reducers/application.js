import { buildActions } from 'utils';

export const types = buildActions('application', [
  'INIT_APP',
  'LOGIN_USER',
  'LOGIN_USER_SUCCESS',
  'LOGIN_USER_FAILURE',
  'LOGOUT_USER',
  'LOGOUT_USER_SUCCESS',
  'LOGOUT_USER_FAILURE',
  'REGISTER_USER',
  'REGISTER_USER_SUCCESS',
  'REGISTER_USER_FAILURE'
]);

const initApp = () => ({
  type: types.INIT_APP
});

const loginUser = (emailAddress, password) => ({
  type: types.LOGIN_USER,
  emailAddress,
  password
});

const loginUserSuccess = user => ({
  type: types.LOGIN_USER_SUCCESS,
  user
});

const loginUserFailure = error => ({
  type: types.LOGIN_USER_FAILURE,
  error
});

const logoutUser = () => ({
  type: types.LOGOUT_USER
});

const logoutUserSuccess = () => ({
  type: types.LOGOUT_USER_SUCCESS
});

const logoutUserFailure = error => ({
  type: types.LOGOUT_USER_FAILURE,
  error
});

const registerUser = details => ({
  type: types.REGISTER_USER,
  details
});

const registerUserSuccess = () => ({
  type: types.REGISTER_USER_SUCCESS
});

const registerUserFailure = error => ({
  type: types.REGISTER_USER_FAILURE,
  error
});

export const actions = {
  initApp,
  loginUser,
  loginUserSuccess,
  loginUserFailure,
  logoutUser,
  logoutUserSuccess,
  logoutUserFailure,
  registerUser,
  registerUserSuccess,
  registerUserFailure
};

export const initialState = {
  loggingIn: false,
  loggingOut: false,
  user: null,
  error: null,
  registration: {
    registering: false,
    complete: false,
    error: null
  }
};

export const reducer = (state = initialState, action = {}) => {
  switch (action.type) {
    case types.LOGIN_USER:
      return {
        ...state,
        loggingIn: true
      };
    case types.LOGOUT_USER:
      return {
        ...state,
        loggingOut: true
      };
    case types.REGISTER_USER:
      return {
        ...state,
        registration: {
          ...state.registration,
          registering: true,
          details: action.details
        }
      };
    case types.LOGOUT_USER_SUCCESS:
      return {
        ...state,
        loggingOut: false,
        user: null
      };
    case types.LOGIN_USER_SUCCESS:
      return {
        ...state,
        loggingIn: false,
        user: action.user
      };
    case types.LOGIN_USER_FAILURE:
    case types.LOGOUT_USER_FAILURE:
      return {
        ...state,
        loggingIn: false,
        loggingOut: false,
        error: action.error
      };
    case types.REGISTER_USER_SUCCESS:
      return {
        ...state,
        registration: {
          registering: false,
          complete: true,
          error: null
        }
      };
    case types.REGISTER_USER_FAILURE:
      return {
        ...state,
        registration: {
          registering: false,
          complete: true,
          error: action.error
        }
      };
    default:
      return state;
  }
};
