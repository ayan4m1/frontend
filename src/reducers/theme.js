import { buildActions } from 'utils';

export const types = buildActions('theme', ['SET_THEME', 'TOGGLE_DARK_MODE']);

const setTheme = theme => ({
  type: types.SET_THEME,
  theme
});

const toggleDarkMode = () => ({
  type: types.TOGGLE_DARK_MODE
});

export const actions = {
  setTheme,
  toggleDarkMode
};

export const initialState = {
  name: 'default'
};

export const reducer = (state = initialState, action = {}) => {
  switch (action.type) {
    case types.SET_THEME:
      return {
        ...state,
        name: action.theme
      };
    case types.TOGGLE_DARK_MODE:
      if (state.name === 'default') {
        return {
          ...state,
          name: 'dark'
        };
      } else {
        return {
          ...state,
          name: 'default'
        };
      }
    default:
      return state;
  }
};
