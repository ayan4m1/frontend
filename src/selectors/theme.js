import { createSelector } from 'reselect';

export const getTheme = state => state.theme;

export const getCurrentTheme = createSelector(getTheme, theme => theme.theme);
