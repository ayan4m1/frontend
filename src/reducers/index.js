import { combineReducers } from 'redux';
import {
  reducer as application,
  initialState as applicationState
} from './application';
import {
  reducer as dashboard,
  initialState as dashboardState
} from './dashboard';
import { reducer as flavor, initialState as flavorState } from './flavor';
import { reducer as flavors, initialState as flavorsState } from './flavors';
import { reducer as profile, initialState as profileState } from './profile';
import { reducer as recipe, initialState as recipeState } from './recipe';
import { reducer as roles, initialState as rolesState } from './roles';
import { reducer as toast, initialState as toastState } from './toast';
import { reducer as users, initialState as usersState } from './users';
import { reducer as theme, initialState as themeState } from './theme';

export const initialState = {
  application: applicationState,
  dashboard: dashboardState,
  flavor: flavorState,
  flavors: flavorsState,
  profile: profileState,
  recipe: recipeState,
  roles: rolesState,
  toast: toastState,
  users: usersState,
  theme: themeState
};

export default combineReducers({
  application,
  dashboard,
  flavor,
  flavors,
  profile,
  recipe,
  roles,
  toast,
  users,
  theme
});
