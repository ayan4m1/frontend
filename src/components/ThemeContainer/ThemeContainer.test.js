import React from 'react';
import renderer from 'react-test-renderer';
import ConnectedThemeContainer, { ThemeContainer } from './ThemeContainer';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import { initialState } from 'reducers/theme';

describe('<ThemeContainer />', () => {
  const props = {
    theme: 'default'
  };

  const mockStore = configureStore();
  const store = mockStore({ theme: initialState });

  it('renders correctly', () => {
    const component = renderer.create(<ThemeContainer {...props} />);

    expect(component.toJSON()).toMatchSnapshot();
  });

  it('renders connected component correctly', () => {
    const component = renderer.create(
      <Provider store={store}>
        <ConnectedThemeContainer />
      </Provider>
    );

    expect(component.toJSON()).toMatchSnapshot();
  });

  it('handles componentDidUpdate correctly', () => {
    const component = renderer.create(<ThemeContainer {...props} />);

    component.update(<ThemeContainer theme="dark" />);

    expect(document.body.classList.item(0)).toEqual('theme--dark');
    expect(document.body.classList.length).toEqual(1);

    component.update(<ThemeContainer theme="default" />);
    expect(document.body.classList.item(0)).toEqual('theme--default');
    expect(document.body.classList.length).toEqual(1);
  });
});
