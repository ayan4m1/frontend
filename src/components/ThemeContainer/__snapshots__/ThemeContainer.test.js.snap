// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ThemeContainer /> renders connected component correctly 1`] = `null`;

exports[`<ThemeContainer /> renders correctly 1`] = `null`;
